executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -92,15 +108,25 @@ pub fn split_out_machines<'a, T: FieldElement>(
         let connecting_identities = identities
             .iter()
             .cloned()
-            .filter(|i| {
-                refs_in_selected_expressions(&i.right)
-                    .intersection(&machine_witnesses)
-                    .next()
-                    .is_some()
+            .filter_map(|i| {
+                let id = i.id();
+                match i {
+                    Identity::Polynomial(_) => None,
```

Comment:
Maybe extract that into a function?

Reply:
simplified due to refactor of `ConnectingIdentity`